lib/ood_core/job/adapters/slurm.rb

```diff
@@ -89,22 +95,66 @@ def initialize(cluster: nil, bin: nil, conf: nil, bin_overrides: {})
           #   #  ...
           #   #]
           # @param id [#to_s] the id of the job
-          # @param filters [Array<Symbol>] list of attributes to filter on
+          # @param attrs [Array<Symbol>, nil] list of attributes request when calling squeue
           # @raise [Error] if `squeue` command exited unsuccessfully
           # @return [Array<Hash>] list of details for jobs
-          def get_jobs(id: "", filters: [])
-            delim = "\x1F"     # don't use "|" because FEATURES uses this
-            options = filters.empty? ? fields : fields.slice(*filters)
-            args  = ["--all", "--states=all", "--noconvert"]
-            args += ["-o", "#{options.values.join(delim)}"]
-            args += ["-j", id.to_s] unless id.to_s.empty?
-            lines = call("squeue", *args).split("\n").map(&:strip)
+          def get_jobs(id: "", attrs: nil, record_separator: RECORD_SEPARATOR, unit_separator: UNIT_SEPARATOR)
+            fields = squeue_fields(attrs)
+            args = squeue_args(id: id, options: fields.values, record_separator: record_separator, unit_separator: unit_separator)
+
+            #TODO: switch mock of Open3 to be the squeue mock script
```

Comment:
Note that the "switch mock of Open3 to be the squeue mock script" is not in your TODO list.

Reply:
Added. The idea is to add a script to a spec/fixtures/ directory that outputs https://github.com/OSC/ood_core/blob/4ee4b03e9997e5381016ee2742e053fe29d99fda/spec/job/adapters/slurm_spec.rb#L260-L262 so we can use bin_overrides instead of doing

```
allow(Open3).to receive(:capture3).and_return([squeue_output, "", double("success?" => true)])
```